src/sqlfluff/dialects/dialect_materialize.py

```diff
@@ -0,0 +1,1016 @@
+            Ref("AlterSourceSinkSizeStatementSegment"),
+            Ref("CloseStatementSegment"),
+            Ref("CopyToStatementSegment"),
+            Ref("CopyFromStatementSegment"),
+            Ref("CreateClusterStatementSegment"),
+            Ref("CreateClusterReplicaStatementSegment"),
+            Ref("CreateConnectionStatementSegment"),
+            Ref("CreateIndexStatementSegment"),
+            Ref("CreateMaterializedViewStatementSegment"),
+            Ref("CreateSecretStatementSegment"),
+            Ref("CreateSinkKafkaStatementSegment"),
+            Ref("CreateSourceKafkaStatementSegment"),
+            Ref("CreateSourceLoadGeneratorStatementSegment"),
+            Ref("CreateSourcePostgresStatementSegment"),
+            Ref("CreateTypeStatementSegment"),
+            Ref("CreateViewStatementSegment"),
+            Ref("DropStatementSegment"),
+            Ref("FetchStatementSegment"),
+            Ref("MaterializeExplainStatementSegment"),
+            Ref("ShowStatementSegment"),
+            Ref("ShowCreateStatementSegment"),
+            Ref("ShowIndexesStatementSegment"),
+            Ref("ShowMaterializedViewsStatementSegment"),
+            Ref("DeclareStatementSegment"),
+        ],
+        remove=[
+            Ref("CreateIndexStatementSegment"),
+            Ref("DropIndexStatementSegment"),
+        ],
+    )
+
+materialize_dialect.sets("materialize_sizes").clear()
+materialize_dialect.sets("materialize_sizes").update(
+    [
+        "3xsmall",
+        "2xsmall",
+        "xsmall",
+        "small",
+        "medium",
+        "large",
+        "xlarge",
+        "2xlarge",
+        "3xlarge",
+        "4xlarge",
+        "5xlarge",
+        "6xlarge",
+    ],
+)
+
+materialize_dialect.add(
+        InstanceSizes=OneOf(
+        MultiStringParser(
+            materialize_dialect.sets("materialize_sizes"),
+            KeywordSegment,
+            type="materialize_size",
+        ),
+        MultiStringParser(
+            [
+                f"'{compression}'"
+                for compression in materialize_dialect.sets("materialize_sizes")
+            ],
+            KeywordSegment,
+            type="compression_type",
+        ),
+    ),
+)
+
+class AlterConnectionRotateKeys(BaseSegment):
+    """`ALTER CONNECTION` statement.
+    """
+    type = "alter_connection_rotate_keys"
+
+    match_grammar = Sequence(
+        "ALTER",
+        "CONNECTION",
+        Ref("IfExistsGrammar", optional=True),
+        Ref("ObjectReferenceSegment"),
+        "ROTATE",
+        "KEYS"
+    )
+
+class AlterRenameStatementSegment(BaseSegment):
+    """A `ALTER RENAME` statement.
+    """
+    _object_types = OneOf(
+        "CONNECTION",
+        "INDEX",
+        "SOURCE",
+        "SINK",
+        "VIEW",
+        Sequence("MATERIALIZED", "VIEW"),
+        "TABLE",
+        "SECRET"
+    )
+
+    type = "alter_rename_statement"
+
+    match_grammar = Sequence(
+        "ALTER",
+        _object_types,
```

Comment:
We usually like just in-lining all the grammar instead of breaking into a new variable like `_object_types`. If important to note a certain section, we could put a comment line before it.

Same comment a few other places in this file.

Reply:
Thank you for taking the time to review this! I've made all of the requested changes. Any other feedback is really appreciated!